### PR TITLE
[testkit] project import test

### PR DESCRIPTION
**Pull request by Abrams27**

### CODE REVIEW

**Comment by andrefmrocha** [comment]
Thanks for this! I had a small question regarding this addition you're trying to make tho. Shouldn't this be an extension to the `resolveProject` test? The only real difference between what they are both doing is that `resolveProject` won't run language specific requests, but maybe we should just add those options there, instead of using another request. Wdyt?

**Comment by Abrams27** [comment]
> Thanks for this! I had a small question regarding this addition you're trying to make tho. Shouldn't this be an extension to the `resolveProject` test? The only real difference between what they are both doing is that `resolveProject` won't run language specific requests, but maybe we should just add those options there, instead of using another request. Wdyt?

well, tbh i don't know, but you're right, will fix it

**Comment by andrefmrocha** [comment]
This request leads me to think maybe this full import request should have an api for making the logic a should have to do, if a java source or a scala source are found, the `javac` or `scalac` requests should happen. Do you think it would make sense for that to be in the scope of this PR and should I create an issue for me to work on it myself?

**Comment by Abrams27** [comment]
> This request leads me to think maybe this full import request should have an api for making the logic a should have to do, if a java source or a scala source are found, the `javac` or `scalac` requests should happen. Do you think it would make sense for that to be in the scope of this PR and should I create an issue for me to work on it myself?

I think that current solution with flags is more flexible and allows developer to decide what to test. Also, it makes this change backward compatible

File: testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/TestClient.scala [hunk 1]
```diff
@@ -800,6 +781,91 @@ class TestClient(
 
   def testWorkspaceReload(): Unit =
     wrapTest(testWorkspaceReload)
+
+  def testResolveProject(javacOptionsFlag: Boolean = false, scalacOptionsFlag: Boolean = false): Unit =
+    wrapTest(testResolveProject(_, javacOptionsFlag, scalacOptionsFlag))
+
+  private def testResolveProject(session: MockSession,
```
[the remark this answers is not in the document]
Comment (Abrams27) [reply testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/TestClient.scala:788]: tbh i wasn't sure why those methods are public - as a testkit client i cannot use them properly, cuz i mocksession is testkit internal thing (am i right?).

[the remark this answers is not in the document]
Comment (andrefmrocha) [reply testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/TestClient.scala:788]: There is a method that allows you to make your own interactions with the testkit, in order to make custom tests. There's a test that exemplifies it, it's the one you had to change in this PR

[the remark this answers is not in the document]
Comment (Abrams27) [reply testkit/src/main/scala/ch/epfl/scala/bsp/testkit/client/TestClient.scala:788]: ohhh okey, I misunderstood it, sorry

**Review by andrefmrocha — APPROVED** [review]
Lgtm! Good work!

**Review by jastice — APPROVED** [review]

**Pull request merged**